src/libs/metadata_view.c

```diff
@@ -1168,6 +1168,19 @@ static void _set_ellipsize(GtkTreeViewColumn *col, GtkCellRenderer *renderer,
   g_object_set(renderer, "ellipsize", ellipsize, NULL);
 }
 
+static gboolean _view_redraw(GtkWidget *view, cairo_t *cr, dt_lib_module_t *self)
+{
+  GtkTreeViewColumn *col0 = gtk_tree_view_get_column(GTK_TREE_VIEW(view), 0);
+  GtkTreeViewColumn *col1 = gtk_tree_view_get_column(GTK_TREE_VIEW(view), 1);
+  const int width0 = gtk_tree_view_column_get_width(col0);
+  const int width1 = gtk_tree_view_column_get_width(col1);
+  // keep 1/3-2/3 ratio
+  const int w0 = (width0 + width1) / 3;
+  // strange. The logic would be to apply it on col0, but only works the other way
```

Comment:
What is happening here, is that you are allocating a fixed "minimum" width of 1/3 of the column width to the second column. The remainder is being equally split with the first column. So they end up with 1/3 and 2/3. Without that, and with both columns being ellipsized, their minimum widths would have been 0 and they would both get 1/2 of the "remainder".

This is not dependent on the content, so you don't have to do it with every redraw. A faster way is:
```
static void _view_size(GtkWidget *view, GdkRectangle *allocation, GtkWidget *header)
{
  gtk_tree_view_column_set_fixed_width(gtk_tree_view_get_column(GTK_TREE_VIEW(view), 1), allocation->width / 3);
}
```
and connect it below with `g_signal_connect(G_OBJECT(view), "size-allocate", G_CALLBACK(_view_size), self);`

The main difference with previously is indeed that it doesn't depend on content, so the first column could still become larger than needed, wasting space, and the second doesn't get more space if it contains longer strings (or give up space if all the strings are short). But this is an improvement anyway.

Reply:
Sorry, didn't realise this was already merged. @phweyland could you please submit again with my suggestion above?

Reply:
> you are allocating a fixed "minimum" width of 1/3 of the column width to the second column. 

This is not obvious given the routine which is used, gtk_tree_view_column_set_fixed_width(). However I've got the same result with gtk_tree_view_column_set_min_width().

> The remainder is being equally split with the first column. So they end up with 1/3 and 2/3. 

Thanks. This is cristal clear now !


> so you don't have to do it with every redraw.

Agreed. I'll make the change (sorry @TurboGit , one PR more to be merges)

Reply:
> This is not obvious

Is anything in Gtk? ;-)

Reply:
BTW if you're interested, Gtk also has a concept of _natural_ width, i.e. the width before ellipsizing. So even if a column can be ellipsized and therefore its _minimum_ width is zero, a grid uses the proportions of _natural_ width to allocate "leftover" space. Why that isn't done for treeviews (and notebooks and probably a bunch of other widgets) is beyond me, but that's how it is.

Reply:
>  Gtk also has a concept of _natural_ width, i.e. the width before ellipsizing.

Some progress. I'm able to get the natural width (which depends on the visible metadata).
So the idea is to use the current rule as long the the view width < 3 * the column 0 natural width, and to keep the column 0 at natural width above. But I'm still failing forcing the column 0 to natural width.